cirq-core/cirq/sim/clifford/act_on_clifford_tableau_args.py

```diff
@@ -94,24 +68,105 @@ def sample(
         # Unnecessary for now but can be added later if there is a use case.
         raise NotImplementedError()
 
-
-def _strat_act_on_clifford_tableau_from_single_qubit_decompose(
-    val: Any, args: 'cirq.ActOnCliffordTableauArgs', qubits: Sequence['cirq.Qid']
-) -> bool:
-    if num_qubits(val) == 1:
-        if not has_unitary(val):
-            return NotImplemented
-        u = unitary(val)
-        clifford_gate = SingleQubitCliffordGate.from_unitary(u)
-        if clifford_gate is not None:
-            for axis, quarter_turns in clifford_gate.decompose_rotation():
-                if axis == pauli_gates.X:
-                    common_gates.XPowGate(exponent=quarter_turns / 2)._act_on_(args, qubits)
-                elif axis == pauli_gates.Y:
-                    common_gates.YPowGate(exponent=quarter_turns / 2)._act_on_(args, qubits)
-                else:
-                    assert axis == pauli_gates.Z
-                    common_gates.ZPowGate(exponent=quarter_turns / 2)._act_on_(args, qubits)
-            return True
-
-    return NotImplemented
+    def _x(self, g: common_gates.XPowGate, axis: int):
+        exponent = g.exponent
+        if exponent % 2 == 0:
+            return
+        assert exponent % 0.5 == 0.0
```

Comment:
Avoid assertions outside of test code - prefer raising an error with a useful debugging message.

Reply:
This *should* never trigger though. It's a private method only called by `_strat_apply_gate`, which tests `has_stabilizer_effect` first. Are we allowed to keep assertions in those cases, or do those need to change to ValueError too (or remove it entirely)?

Reply:
Our style guide places no restriction on assert vs raise-error, so this is optional. That said...

My take is that a well-written error message is much easier to debug than an `assert`, and on a long enough time scale:
1. even "never touched" code will trigger, and
2. the original authors won't be available to help debug